x-pack/platform/plugins/shared/streams/server/lib/tasks/storage.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { types, type IndexStorageSettings, type IStorageClient } from '@kbn/storage-adapter';
+import type { PersistedTask } from './types';
+
+export const taskStorageSettings = {
+  name: '.kibana_streams_tasks',
+  schema: {
+    properties: {
+      // Do I really want to index this?
```

Comment:
These kind of storage blocks are for me like using a relational database. So the fields for retrieval should be indexed. Everything you need to query on, doesn't (like payload)